packages/hoppscotch-common/src/components/collections/ImportExport.vue

```diff
@@ -1,361 +1,430 @@
+      } else {
+        showImportFailedError()
+      }
+    },
+  }),
+}
 
-const inputChooseFileToImportFrom = ref<HTMLInputElement | any>()
-const mySelectedCollectionID = ref<number | undefined>(undefined)
-const inputChooseGistToImportFrom = ref<string>("")
+const HoppMyCollectionImporter: ImporterOrExporter = {
+  metadata: {
+    id: "hopp_my_collection",
+    name: "import.from_my_collections",
+    title: "import.from_my_collections_description",
+    icon: IconUser,
+    disabled: false,
+    applicableTo: ["team-workspace"],
+  },
+  component: defineStep("my_collection_import", MyCollectionImport, () => ({
+    async onImportFromMyCollection(content) {
+      const { collectionsType } = props
 
-const importerModules = computed(() =>
-  RESTCollectionImporters.filter(
-    (i) => i.applicableTo?.includes(props.collectionsType) ?? true
-  )
-)
+      const isTeamCollection = collectionsType.type === "team-collections"
 
-const importerModule = computed(() => {
-  if (importerType.value === null) return null
-  return importerModules.value[importerType.value]
-})
+      const hasTeamWriteAccess = isTeamCollection
+        ? collectionsType.selectedTeam?.myRole == "EDITOR" ||
+          collectionsType.selectedTeam?.myRole == "OWNER"
+        : false
 
-const importerSteps = computed(() => importerModule.value?.steps ?? null)
+      if (!hasTeamWriteAccess) {
+        showImportFailedError()
+        return
+      }
 
-const enableImportButton = computed(
-  () => !(stepResults.value.length === importerSteps.value?.length)
-)
+      isTeamCollection &&
+        collectionsType.selectedTeam?.id &&
+        toTeamsImporter(content, collectionsType.selectedTeam?.id)()
+    },
+  })),
+}
 
-watch(mySelectedCollectionID, (newValue) => {
-  if (newValue === undefined) return
-  stepResults.value = []
-  stepResults.value.push(newValue)
-})
+const HoppOpenAPIImporter: ImporterOrExporter = {
+  metadata: {
+    id: "hopp_openapi",
+    name: "import.from_openapi",
+    title: "import.from_openapi_description",
+    icon: IconOpenAPI,
+    disabled: false,
+    applicableTo: ["personal-workspace", "url-import"],
+  },
+  supported_sources: [
+    {
+      id: "file_import",
+      name: "import.from_file",
+      icon: IconFile,
+      step: FileSource({
+        caption: "import.from_file",
+        acceptedFileTypes: ".json",
+        onImportFromFile: async (content) => {
+          const res = await hoppOpenAPIImporter(content)()
 
-watch(inputChooseGistToImportFrom, (url) => {
-  stepResults.value = []
-  if (url === "") {
-    hasGist.value = false
-  } else {
-    hasGist.value = true
-    stepResults.value.push(inputChooseGistToImportFrom.value)
-  }
-})
+          if (E.isRight(res)) {
+            onSuccessfulImport(res.right)
+          } else {
+            showImportFailedError()
+          }
+        },
+      }),
+    },
+    {
+      id: "url_import",
+      name: "import.from_url",
+      icon: IconFile,
```

Comment:
```suggestion
      icon: IconLink,
```
we can use a link icon rather than a file here